packages/msal-interactor/src/MsalInteractor.ts

```diff
@@ -2,105 +2,230 @@
  * Copyright (c) Microsoft. All rights reserved.
  * Licensed under the MIT license. See LICENSE file in the project.
  */
+import type {
+	AccountInfo,
+	AuthenticationResult,
+	Configuration,
+	IPublicClientApplication,
+} from '@azure/msal-browser'
 import {
-	UserAgentApplication,
-	AuthResponse,
 	InteractionRequiredAuthError,
-	Configuration,
-} from 'msal'
+	PublicClientApplication,
+	SilentRequest,
+} from '@azure/msal-browser'
+
+export interface MsalInteractorOptions {
+	msalConfig: Configuration
+	oidcScopes: string[]
+}
+
+export interface MsalInteractorLoginOptions {
+	usePopup?: boolean
+}
 
 export class MsalInteractor {
-	private _instance: UserAgentApplication | undefined
+	private static _instance: MsalInteractor | null
+	private static _options: MsalInteractorOptions | null
+	private _msalConfig: Configuration
+	private _oidcScopes: string[]
+	private _msalInstance: IPublicClientApplication
+	private _accountInfo: AccountInfo | null = null
+	public isAuthenticated: () => Promise<boolean>
 
-	public constructor(
-		private msalConfig: Configuration,
-		private scopes: string[],
-		private authDisabled = false,
-	) {}
+	/**
+	 * Singleton. Use MsalInteractor.configure(options)
+	 * and MsalInteractor.getInstance()
+	 */
+	private constructor({ msalConfig, oidcScopes }: MsalInteractorOptions) {
+		this._msalConfig = msalConfig
+		this._oidcScopes = oidcScopes
+		this._msalInstance = new PublicClientApplication(this._msalConfig)
+
+		// On page load/instantiation, check if responding to redirect auth flow
+		const pageLoadHandler = this._msalInstance
+			.handleRedirectPromise()
+			.then(this._handleAuthResponse)
+
+		this.isAuthenticated = () => pageLoadHandler.then(res => !!res)
+	}
+
+	public static configure = (options: MsalInteractorOptions): void => {
```

Comment:
The static configure() method is a little gross, you end up with duplicated state statically and as a class field. Additionally this makes the API less flexible than just calling `new MsalInteractor(config)`. I think it would be more straightforward as a plain constructor.

Reply:
I cleaned up the duplicate static and instance state. 

I went with a singleton pattern because there is an issue if one instantiates msal more than once or register multiple handleRedirectPromise listeners then the app gets stuck in a redirect loop when using the redirect flow as subsequent handleRedirectPromise fire too soon before the cache is populated. I used a singleton pattern to get around this issue. I cleaned up the duplicate state but if preferred, I can refactor this to use a public constructor and rely on those consuming this package to just instantiate it once per application. I am able to replicate this problem but have not found it documented in the @azure/msal-browser repo so I plan to open an issue there.

Reply:
Ahh, gotcha. Yeah, let's put a comment on the class about how this should be instantiated as a singleton in apps, but leave it as a public constructor. If this becomes problematic we can add a factory module alongside the class module.

Reply:
To follow up on this, can you make sure the comment is a JsDoc on the class itself in addition to the example you have in the docs? It's handy to have info like that available when you're in vscode.

Reply:
I added a JSDOC to the class that matches what I had in the documentation.